core/contract/src/main/java/org/eclipse/dataspaceconnector/contract/ContractServiceExtension.java

```diff
@@ -118,18 +119,21 @@ private void registerServices(ServiceExtensionContext context) {
 
         var waitStrategy = context.hasService(NegotiationWaitStrategy.class) ? context.getService(NegotiationWaitStrategy.class) : new ExponentialWaitStrategy(DEFAULT_ITERATION_WAIT);
 
+        ContractNegotiationObservableImpl observable = new ContractNegotiationObservableImpl();
```

Comment:
The observable should also be registered in the context.

Reply:
agree. done